draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,607 @@
+
+   The NS RRset and glue, in the response of the legacy query that was sent in parallel to the incremental deleg query, MUST NOT be used, but the signed DS record (or NSEC(3) records indicating that there was no DS) MUST be used in linking the DNSSEC authentication chain as which would conventionally be done with DNSSEC as well.
+
+2. The incremental deleg query name does not exist (NXDOMAIN).
+
+   There is no incremental delegation for the subzone, and the referral response for the legacy delegation MUST be processed  as would be done with legacy DNS and DNSSEC processing.
+
+3. The incremental deleg query name does exist, but resulted in an NOERROR no answer responses (also known as a NODATA response).
+
+   If the legacy query, did result in a referral for the same number of labels as the subdomain that the incremental deleg query was for, then there was no incremental delegation for the subzone, and the referral response for the legacy delegation MUST be processed as would be done with legacy DNS and DNSSEC processing.
+
+   Otherwise, the subzone may be more than one label below the delegating zone.
+
+   If the response to the legacy query resulted in a referral, then a new incremental deleg query MUST be spawned, matching the zone cut of the legacy referral response.
+   For example if the triggering query is `www.university.ac.example.` and the target zone `example.`, and the legacy response contained an NS RRset for `university.ac.example.`, then the incremental deleg query name is `university.ac._deleg.example.`
+   The response to the new incremental deleg query MUST be processed as described above, as if it was the initial incremental deleg query.
+
+   If the legacy query was send minimised and needs a followup query, then parallel to that followup query a new incremental deleg query will be send, adding a single label to the previous incremental deleg query name.
+   For example if the triggering query is `www.university.ac.example.` and the target zone `example.` and the minimised legacy query name was `ac.example.` (which also resulted in a NOERROR no answer response), then the incremental deleg query to be send along in parallel with the followup legacy query will become `university.ac.example.`
+   Processing of the responses of those two new queries will as described above.
+
+## `_deleg` label presence {#presence}
+
+Absence of the `_deleg` label in a zone is a clear signal that the zone does not contain any incremental deleg delegations.
+Recursive resolvers SHOULD NOT send any additional incremental deleg queries for zones for which it is known that it does not contain the `_deleg` label at the apex.
+The state regarding the presence of the `_deleg` label within a resolver can be "unknown", "known not to be present", or "known to be present".
+
+When the presence of a `_deleg` label is "unknown", a `_deleg` presence test query SHOULD be send in parallel to the next query for the zone.
```

Comment:
Here it says that a _deleg presence query should be be send when the presence of _deleg is unknown, but in table 1 it says that when there is deleg support in the nameserver no additional query needs to be send even when the _deleg presence is unknown. so i think this statement needs an extra clause that states that this is not needed when it is known that there is nameserver support.